pkg/authenticator/config/config.go

```diff
@@ -28,12 +29,37 @@ const (
 	DefaultClientCertPath = "/etc/conjur/ssl/client.pem"
 	DefaultTokenFilePath  = "/run/conjur/access-token"
 
+	DefaultConjurVersion = "5"
+
 	// DefaultTokenRefreshTimeout is the default time the system waits to reauthenticate on error
 	DefaultTokenRefreshTimeout = 6 * time.Minute
 )
 
+// NewFromEnvFunc defines the method for creating a new config from the env
+type NewFromEnvFunc func(Options) (*Config, error)
+// ReadSSLCertFunc defines the method for reading an SSL Certificate from the env
+type ReadSSLCertFunc func() ([]byte, error)
+// NewUserNameFunc defines the method for parsing a username from the env
+type NewUserNameFunc func(string) (*Username, error)
+// Options contains the functional options needed for NewFromEnv
+type Options struct {
+	ReadSSLCert ReadSSLCertFunc
+	NewUserName NewUserNameFunc
+}
+
+// DefaultConfigOptions implement the methods defined within config.go
+var DefaultConfigOptions = Options{
+	readSSLCert,
+	NewUsername,
+}
+
+// DefaultNewFromEnv returns NewFromEnv with the default functional options
+func DefaultNewFromEnv() (*Config, error) {
+	return NewFromEnv(DefaultConfigOptions)
+}
+
 // New returns a new authenticator configuration object
-func NewFromEnv() (*Config, error) {
+func NewFromEnv(opts Options) (*Config, error) {
```

Comment:
This is a breaking change in almost all packages that use our authenticator. I would highly avoid changing this method's signature. As far as I can tell, this was changed to accommodate the tests but this can be tested without this change I think.